src/types/PostgresErrorCode.ts

```diff
@@ -1,6 +1,20 @@
 // See https://www.postgresql.org/docs/8.2/errcodes-appendix.html
 export enum PostgresErrorCode {
-  FOREIGN_KEY_VIOLATION = '23503',
-  UNIQUE_VIOLATION = '23505',
-  INSUFFICIENT_RESOURCES = '53000',
+  FOREIGN_KEY_VIOLATION,
+  UNIQUE_VIOLATION,
+  INSUFFICIENT_RESOURCES,
+  UNKNOWN_ERROR,
 }
+
+export const postgresErrorCodeStringToEnum = (errorCode: string): PostgresErrorCode => {
+  switch (errorCode) {
+    case '23503':
+      return PostgresErrorCode.FOREIGN_KEY_VIOLATION;
+    case '23505':
+      return PostgresErrorCode.UNIQUE_VIOLATION;
+    case '53000':
+      return PostgresErrorCode.INSUFFICIENT_RESOURCES;
+    default:
+      return PostgresErrorCode.UNKNOWN_ERROR;
```

Comment:
If we wanna be pedantic this might not be an *unknown* error but rather is just an "other" error.  I'm gonna change to reflect that.